src/openforms/api/serializers.py

```diff
@@ -85,15 +112,79 @@ def get_child_serializer_class(self):
     def process_object(self, obj):
         return obj
 
-    def create(self, validated_data):
-        model = self.get_child_serializer_class().Meta.model
-
-        objects_to_create = []
-        for data_dict in validated_data:
-            obj = model(**data_dict)
-            objects_to_create.append(self.process_object(obj))
+    def _batch(
+        self,
+        data: Iterable[MutableMapping[str, Any]],
+    ) -> Iterable[BatchItem]:
+        # Helps create() and validate() deal with nested serializers in O(1) db
+        # queries. i.e. one insert per model type.
+        #
+        # It helps by yielding BatchItems of unsafed Model instances. Nested
+        # models should be safed first, so parent gets saved with the new
+        # foreign keys. The order of BatchItems is defined such that high
+        # priority comes first, and models of the same type are grouped
+        # together for a single bulk_create.
```

Comment:
Is this clear?
I initially had this as recursive function, so children could have lists of children too. But we don't need that (yet/ever?) and this is a lot simpler to follow.